lib/moonshot/stack_template.rb

```diff
@@ -10,30 +8,19 @@ def required?
       end
     end
 
-    attr_reader :body
+    BODY_MAX_LENGTH_IN_BYTES = 51_200
 
     def initialize(filename, log:)
       @log = log
-
-      unless File.exist?(filename)
-        @log.error("Could not find CloudFormation template at #{filename}")
-        raise
-      end
-
-      # The maximum TemplateBody length is 51,200 bytes, so we remove
-      # formatting white space.
-      @body = JSON.parse(File.read(filename)).to_json
+      @filename = Dir[filename].first
     end
 
-    def parameters
-      JSON.parse(@body).fetch('Parameters', {}).map do |k, v|
-        Parameter.new(k, v['Default'])
-      end
+    def validate
+      @body.bytes.length <= MAX_TEMPLATE_BODY_LENGTH_IN_BYTES
     end
 
-    # Return a list of defined resource names in the template.
-    def resource_names
-      JSON.parse(@body).fetch('Resources', {}).keys
+    def exists?
```

Comment:
I think we should use `#exist?` to align with the method names in stdlib, Ruby has been deprecating `#exists?`.